mapie/tests/test_quantile_regression.py

```diff
@@ -354,14 +399,41 @@ def test_invalid_cv(cv: Any) -> None:
         match=r".*Invalid cv method.*",
     ):
         mapie = MapieQuantileRegressor(cv=cv)
-        mapie.fit(X_train_toy, y_train_toy, X_calib_toy, y_calib_toy)
+        mapie.fit(
+            X_train_toy,
+            y_train_toy,
+            X_calib=X_calib_toy,
+            y_calib=y_calib_toy
+        )
 
 
 @pytest.mark.parametrize("cv", [None, "split"])
 def test_valid_cv(cv: Any) -> None:
     """Test that valid cv raise no errors."""
     mapie = MapieQuantileRegressor(cv=cv)
-    mapie.fit(X_train_toy, y_train_toy, X_calib_toy, y_calib_toy)
+    mapie.fit(
+        X_train_toy,
+        y_train_toy,
+        X_calib=X_calib_toy,
+        y_calib=y_calib_toy
+    )
+
+
+def test_X_calib_isNone() -> None:
+    """Test that the fit method works when X_calib or y_calib is None."""
+    mapie = MapieQuantileRegressor()
+    mapie.fit(X, y, calib_size=0.5)
+    mapie.predict(X)
+
+
+def test_X_calib_isNone_with_sample_weight() -> None:
+    """
+    Test that the fit method works with calib dataset defined is None
+    with sample weights.
+    """
+    mapie = MapieQuantileRegressor()
+    mapie.fit(X, y, calib_size=0.5, sample_weight=np.ones(X.shape[0]))
+    mapie.predict(X)
```

Comment:
Is it possible to test the predictions between 1. a train_test_split done by the user (with X_train_toy, X_calib_toy, y_train_toy, y_calib_toy), 2. a train_test_split performed within MAPIE ?

Reply:
Yes, with implementation of `random_state`, `shuffle` and `stratify` that is now possible. Done.